### PR TITLE
Fix #7042 : Change timescale to tt

### LINKED ISSUE #7042: icrs_to_cirs passed TDB to get_cip
I'm new to AstroPy, but I noticed a discrepancy while reading through the code.

The icrs_to_cirs function converts the observation time to TDB and then calls get_cip:

https://github.com/astropy/astropy/blob/58fdc0115d0204c5bb64a36b5a24939190e4c396/astropy/coordinates/builtin_frames/icrs_cirs_transforms.py#L28-L29

get_cip then calls pnm06a:

https://github.com/astropy/astropy/blob/58fdc0115d0204c5bb64a36b5a24939190e4c396/astropy/coordinates/builtin_frames/utils.py#L146

So get_cip is handing pnm06a time in TDB.  However, the SOFA manual:

http://www.iausofa.org/2017_0420_C/sofa/manual.pdf

Says that pnm06a accepts time in TT.

It is my understanding that TT is an Earth referenced time whereas TDB is a Solar System referenced time.  So the two are not equivalent (though my reading of the SOFA documentation indicates that they are generally within 1.2ms of each other).

Is my understanding incorrect?  Or was passing TDB to pnm06a an optimization of some kind with little measurable difference?  Seems like in most cases when icrs_to_cirs is executed `obstime` will have a UTC source, so if it's already calculating TDB then TT will also be available and so it might as well pass TT to get_cip for the best accuracy.

Thanks.

### PR DESCRIPTION
**Pull request by shreyasbapat**
Fixes #7042 

cc @Juanlu001 

I will add the actual comparison, once I get my astropy up and running :/

### CODE REVIEW

**Comment by shreyasbapat** [comment]
So, I tested this on `J2000` epoch and for `Time.now()` .In J2000, there isn't any difference at all in the results, but when I test it for Time.now(), its a substantial `−0.44157497′′` difference. Here is the code to reproduce the results: 

```
from astropy.coordinates import ICRS, CIRS
from astropy import units as u
from astropy.time import Time

a = CIRS(ra="02h31m48.7s", dec=8*u.deg, distance=2*u.au, obstime=Time.now())

b = a.transform_to(ICRS)

print(b.ra.value)

```

**Comment by shreyasbapat** [comment]
> ts a substantial −0.44157497′′ difference.

Sorry for this one, but the difference is in order of `5.9022787×10−10′′` and it varies with the distance.

**Comment by shreyasbapat** [comment]
@Juanlu001  One test is failing but that does not include the ICRS to CIRS conversion :/ 

That is still definitely me right :/

File: CHANGES.rst [hunk 1]
```diff
@@ -303,6 +303,9 @@ astropy.convolution
 astropy.coordinates
 ^^^^^^^^^^^^^^^^^^^
 
+- Changed the timescale in ICRS to CIRS from 'tdb' to 'tt' conversion and vice-versa,
```
Comment (Juanlu001) [inline CHANGES.rst]: @pllim if I intend to backport this to 2.0.x this changelog entry should be in another place, am I right?

@shreyasbapat, could you please move it?

Comment (shreyasbapat) [reply CHANGES.rst]: Definitely!

**Comment by Juanlu001** [comment]
> One test is failing but that does not include the ICRS to CIRS conversion :/

https://github.com/astropy/astropy/pull/9076 was just merged, can you please rebase to see if the failure goes away?

File: CHANGES.rst (hunk 1 above)
Comment (shreyasbapat) [reply CHANGES.rst]: Hehe, can you please tell me where?

Comment (shreyasbapat) [reply CHANGES.rst]: In 2.0.x's Changelog? Which I suppose not!

Comment (shreyasbapat) [reply CHANGES.rst]: Gotcha!

**Comment by shreyasbapat** [comment]
@Juanlu001 Can you review it?

**Comment by mhvk** [comment]
The change is correct for `get_cip`, but not for `acpi`, which gets called a little later. So, this needs a bit more care - should we get both times or is there one for which the small difference doesn't matter? cc @StuartLittlefair, who wrote most of this.

**Comment by StuartLittlefair** [comment]
I think it depends on how much it slows down the transformation, and the size of the change. The original bug was only of the order of fractions of a micro-arcsecond differences, so it's possible to argue neither time tweak "matters".

**Comment by bsipocz** [comment]
@mhvk @Juanlu001 @eteq - what's the status here? Please review or remilestone to 4.0

**Comment by mhvk** [comment]
My comment above was that there are two erfa routines, which need a different timescale, `pnm06b` needs TT and `acpi` needs TDB. In https://github.com/astropy/astropy/issues/7042#issuecomment-405600536, @StuartLittlefair noted that erfa routines often tell whether one can substitute one for the other, and that `pnm06b` did not note that (though since it is fits with time in centuries, one suspects the difference is tiny). On the other hand, `acpi` does note that "TT can be used instead of TDB without any significant impact on accuracy". That suggests the present PR does the right thing, and we do not have to retrieve both TT and TDB. But I think it would be best if we added a comment right before the line that does `erfa.acpi` to note that that routine requests TDB but is OK with the TT we feed it.

**Comment by shreyasbapat** [comment]
Thanks for the inputs @StuartLittlefair and @mhvk . I will do that in a while.

**Comment by bsipocz** [comment]
@shreyasbapat  - do you plan to come back to it in the next day or so? It's totally cool if not, I'm just asking whether to remilestone or wait for it.

**Comment by shreyasbapat** [comment]
Hi @bsipocz , I am sorry for replying late. I made the changes which @mhvk  asked :D

**Comment by pllim** [comment]
Definitely try a rebase against upstream master.

**Comment by bsipocz** [comment]
Yes, a rebase is needed. And I go ahead and do it now (also remilestoned again as now it's been approved)

**Comment by bsipocz** [comment]
Thanks @shreyasbapat!

**Comment by shreyasbapat** [comment]
Thanks a lot :D

**Pull request merged**